### PR TITLE
Add ZenDesk in the project

### LINKED ISSUE #410: Add ZenDesk widget to footer
As a user I'd like to have an easy way to access Customer Support, search the knowledgebase and open tickets. 

As a customer supporter, I'd like to capture as much user context as possible when tickets are created from the bootcamp site. 

This should be part of the footer on all future pages. It's less important on existing pages, but would be nice.

┆Attachments: <a href="https://github.com/mitodl/bootcamp-ecommerce/pull/420">bootcamp-ecommerce #420 Add ZenDesk in the project</a>

### PR DESCRIPTION
**Pull request by umarmughal824**
#### What are the relevant tickets?
#410 , https://trello.com/c/85UPXytE/82-add-zendesk-widget-to-footer 

#### What's this PR do?
fixes #410, Add Zendesk support for the project

#### How should this be manually tested?
Just open the site, you should be able to see the ZenDesk widget.

#### Where should the reviewer start?
Just set the flag `ZENDESK_HELP_WIDGET_ENABLED=True` in your .env

#### Screenshots
Screenshot 2020-04-29 at 15 04 51

### CODE REVIEW

**Comment by umarmughal824** [comment]
@pdpinch Could you please review the attached screenshots in the PR description.

**Comment by pdpinch** [comment]
Sorry, can we add the help widget, but not change the footer in this PR? I'd like to keep this simple.

**Comment by pdpinch** [comment]
Will this pass the page title as a search term to the widget?

**Comment by umarmughal824** [comment]
Yes it will pass the page title as a search term.
Reference: 
https://github.com/mitodl/bootcamp-ecommerce/pull/420/files#diff-10df0b192047dca7dbbc7a0043668f15R46

**Comment by umarmughal824** [comment]
> Sorry, can we add the help widget, but not change the footer in this PR? I'd like to keep this simple.

@pdpinch Sure will update that. But just a question don't we need that section any more for it or you planning to cover it in a separate ticket?

File: main/settings.py [hunk 1]
```diff
@@ -85,6 +85,16 @@ def get_var(name, default):
...
 SECURE_SSL_REDIRECT = get_var('BOOTCAMP_SECURE_SSL_REDIRECT', True)
 
+ZENDESK_CONFIG = {
+    "HELP_WIDGET_ENABLED": get_bool(
+        "ZENDESK_HELP_WIDGET_ENABLED",
+        False,
+    ),
+    "HELP_WIDGET_KEY": get_string(
+        "ZENDESK_HELP_WIDGET_KEY",
+        "d99f12ec-89dd-4111-b5ea-7b36d01bba24",
```
Comment (asadiqbal08) [inline main/settings.py:95]: I forgot, to which that default value of widget_key is pointing to ?

File: main/templates/base.html [hunk 2]
```diff
@@ -23,11 +23,27 @@
...
+      <script id="ze-snippet" src="https://static.zdassets.com/ekr/snippet.js?key={{zendesk_config.help_widget_key}}"> </script>
+    {% endif %}
   </head>
   <body>
     {% hijack_notification %}
     {% include "gtm_body.html" %}
     {% block content %}
     {% endblock %}
+        {% if zendesk_config.help_widget_enabled %}
+    <script type="text/javascript">
```
Comment (asadiqbal08) [inline main/templates/base.html:36]: nit: indentation.

File: static/js/entry/root.js [hunk 3]
```diff
@@ -7,15 +8,28 @@ import { AppContainer } from "react-hot-loader"
...
 }
 
+const loadZendesk = () => {
+  return <Zendesk zendeskKey={ZENDESK_KEY} />
+}
+
 const renderApp = Component => {
   const history = createBrowserHistory()
+  if (ZENDESK_ENABLED && ZENDESK_KEY) {
+    loadZendesk()
```
Comment (asadiqbal08) [inline static/js/entry/root.js:31]: Are we not just adding zendesk into the footer of bootcamp project ? if yes then why these stuff is the part of this story ? as I remember, It was added to suggest relevant search articles of products.

File: main/settings.py (hunk 1 above)
Comment (umarmughal824) [reply main/settings.py:95]: Javascript one that peter mentioned on the channel.

Comment (umarmughal824) [reply main/settings.py:95]: <script id="ze-snippet" src="https://static.zdassets.com/ekr/snippet.js?key=d99f12ec-89dd-4111-b5ea-7b36d01bba24"> </script>

File: static/js/entry/root.js (hunk 3 above)
Comment (umarmughal824) [reply static/js/entry/root.js:31]: peter has already mentioned it in the ticket to add the search part for it.

**Pull request merged**